clients/pkg/logentry/stages/limit.go

```diff
@@ -34,36 +42,61 @@ func newLimitStage(logger log.Logger, config interface{}, registerer prometheus.
 		return nil, err
 	}
 
+	logger = log.With(logger, "component", "stage", "type", "limit")
+	if cfg.ByLabelName != "" && cfg.MaxDistinctLabels < MinReasonableMaxDistinctLabels {
+		level.Warn(logger).Log(
+			"msg",
+			fmt.Sprintf("max_distinct_labels was adjusted up to the minimal reasonable value of %d", MinReasonableMaxDistinctLabels),
```

Comment:
Any rationale behind not to have `< MinReasonableMaxDistrinctLabels`?

Reply:
> and user need to increase it should they need more.

Just to clarify this bit, user need to increase limit in case if number actively hit ratelimiters exceed it or they'll have innacurate ratelimiting (more entrics will go through then should have). Should some labels values be transient and NOT being in active use (lets say pod_name) , then  it is perfectly fine for the system to reach max limit and get then GCed, there is no need to increase limit to accomodate for them.